upt_macports/upt_macports.py

```diff
@@ -79,7 +79,12 @@ def archive_type(self):
             'xz': 'xz'
         }
         try:
-            archive_name = self.upt_pkg.get_archive().filename
+            if self.upt_pkg.frontend != 'rubygems':
+                archive_name = self.upt_pkg.get_archive().filename
+            else:
+                archive_name = self.upt_pkg.get_archive(
+                    self.upt_pkgArchiveType.RUBYGEM).filename
+
```

Comment:
What you could do here is define an "archive_type" class attribute for all subclasses of MacPortsPackage (just like the "template" attribute) that would be equal to "upt.ArchiveType.RUBYGEM" or "upt.ArchiveType.SOURCE_TARGZ" (note that the latter does not necessarily refer to a tar.gz, because I fucked up when naming it :p).

You would then write:

```python
archive_name = self.upt_pkg.get_archive(self.archive_type).filename
```